### ISSUE TITLE
winrt ObjectObserver::LoadObjectModelAsync will fail if IBuffer goes out of scope

### ISSUE THREAD

**Issue by fieldsJacksonG** [Azure/azure-object-anchors#24]
**Description**

When using the winrt projection and using the Completed event for LoadObjectModelAsync, the IBuffer the input array_view uses will go out of scope before the completed event fires.  Since an array_view is [not a copy](https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/array-view#array_viewarray_view-constructor) of the input data, LoadObjectModelAsync will nullref when the IBuffer goes out of scope.

This gets silently surfaced to the developer as AsyncStatus::Error.

**Steps to reproduce the issue**

```
// AsyncOp is from the Completed event for ReadBufferAsync on the .ou file.
winrt::Windows::Storage::Streams::IBuffer Buffer = AsyncOp.GetResults();

auto LoadObjectTask = ObjectAnchorObserver.LoadObjectModelAsync(
	winrt::array_view(Buffer.data(), Buffer.Length()));
LoadObjectTask.Completed([this](winrt::Windows::Foundation::IAsyncOperation<ObjectModel> AsyncOp,
	winrt::Windows::Foundation::AsyncStatus Status)
{
	// Status == AsyncStatus::Error if the calling function goes out of scope before this event fires.
});
```

**Expected behavior**

I would expect an async function to make a copy of any required data that could go out of scope.

Alternatively, extra logging for failures in this function, or documentation that the IBuffer cannot go out of scope could help devs hitting this in the future.

**Development information (please complete the following information)**

 - Azure Object Anchors SDK version: 0.19.0

**AR Device information (please complete the following information):**

 - Device: HoloLens 2

**Additional context**

This can be worked around by including the IBuffer in LoadObjectModelAsync.Completed's lambda:
`LoadObjectTask.Completed([this, `**`Buffer`**`]`

**Comment by craigktreasure** [comment]
Sorry for the delay, but thank you for the excellent explanation. We've got someone taking a look at it.

Internal reference number is 38627384.

**Comment by craigktreasure** [comment]
Thanks for reporting this issue. It has been fixed in Azure Object Anchors SDK 0.21.0. See the [May 2022 release notes](https://github.com/Azure/azure-object-anchors/releases/tag/2022.5.1-preview).
